connector/solarwindsentityconnector/connector.go

```diff
@@ -24,29 +27,53 @@ import (
 
 type solarwindsentity struct {
 	logsConsumer consumer.Logs
+	entities     map[string]internal.Entity
 
 	component.StartFunc
 	component.ShutdownFunc
 }
 
+var _ connector.Metrics = (*solarwindsentity)(nil)
+var _ connector.Logs = (*solarwindsentity)(nil)
+
 func (s *solarwindsentity) Capabilities() consumer.Capabilities {
 	return consumer.Capabilities{MutatesData: false}
 }
 
 func (s *solarwindsentity) ConsumeMetrics(ctx context.Context, metrics pmetric.Metrics) error {
 	logs := plog.NewLogs()
-	err := s.logsConsumer.ConsumeLogs(ctx, logs)
-	if err != nil {
-		return err
+	events := internal.BuildEventLog(&logs)
+
+	for i := range metrics.ResourceMetrics().Len() {
```

Comment:
this looks incorrect, I guess it should be like this
```suggestion
	for i := 0; i < metrics.ResourceMetrics().Len(); i++
```

Reply:
It is strange, it should be incorrect but it works. However, I have fixed it to the standard form.